torch/csrc/distributed/rpc/rref.cpp

```diff
@@ -199,11 +199,20 @@ template class UserRRef<py::object>;
 
 //////////////////////////  OwnerRRef  /////////////////////////////////////
 
-template <typename T>
-const T& OwnerRRef<T>::getValue() const {
+template <>
+const IValue& OwnerRRef<IValue>::getValue() const {
+  // TODO: use callback to make this non-blocking
+  std::unique_lock<std::mutex> lock(mutex_);
+  valueCV_.wait(lock, [this] { return value_.has_value(); });
+  return value_.value();
+}
+
+template <>
+const py::object& OwnerRRef<py::object>::getValue() const {
   // TODO: use callback to make this non-blocking
   std::unique_lock<std::mutex> lock(mutex_);
   valueCV_.wait(lock, [this] { return value_.has_value(); });
+  PythonRpcHandler::getInstance().handleException(value_.value());
```

Comment:
Is this inconsistent with what we do in `PYTHON_CALL`? In `PYTHON_CALL` the exception is wrapped into payload in `generatePythonUDFResult`, and then the exception message is sent to the caller and thrown there in `toPyObjInternal`. But here, the exception is thrown on the callee (owner) and relies on `RequestCallback::operator()` to catch it and convert that into an exception message? Is this intentional?

Reply:
both PYTHON_CALL and PYTHON_FETCH_CALL re-throw python exception during deserialization. 

Here, callee and caller are in the same process, did not go through deserialization, so skipped python exception re-thrown